addons/mail/static/src/discuss/call/common/channel_member_patch.js

```diff
@@ -17,20 +17,33 @@ const ChannelMemberPatch = {
                 }
                 this.channel_id.rtc_session_ids.add(r);
                 this.store.ringingChannels.add(this.channel_id);
-                this.channel_id.cancelRtcInvitationTimeout = browser.setTimeout(() => {
-                    this.store.env.services["discuss.rtc"].leaveCall(this.channel_id);
-                }, ChannelMember.CANCEL_CALL_INVITE_DELAY);
+                this.startInvitationTimeout();
             },
             /** @this {import("models").ChannelMember} */
             onDelete() {
                 if (!this.channel_id) {
                     return;
                 }
-                browser.clearTimeout(this.channel_id.cancelRtcInvitationTimeout);
+                this.cancelInvitationTimeout();
                 this.store.ringingChannels.delete(this.channel_id);
             },
         });
         this.rtcSession = fields.One("discuss.channel.rtc.session");
     },
+    cancelInvitationTimeout() {
+        if (this.channel_id?.cancelRtcInvitationTimeout) {
+            browser.clearTimeout(this.channel_id.cancelRtcInvitationTimeout);
+            this.channel_id.cancelRtcInvitationTimeout = undefined;
+        }
+    },
+    startInvitationTimeout() {
+        if (this.channel_id.cancelRtcInvitationTimeout) {
+            return;
+        }
+        this.channel_id.cancelRtcInvitationTimeout = browser.setTimeout(() => {
+            this.store.env.services["discuss.rtc"].leaveCall(this.channel_id);
```

Comment:
```suggestion
            this.store.rtc.leaveCall(this.channel_id);
```

_(I think this is old code for when rtc service was not a relational field in store a long time ago)_